test/shared/utils.py

```diff
@@ -111,12 +111,21 @@ def expected_wheels(package_name, package_version):
             '{package_name}-{package_version}-cp38-cp38-win_amd64.whl',
         ]
     elif platform == 'macos':
+        if macosx_deployment_target is not None:
+            tag = macosx_deployment_target.replace(".", "_")
+            tag1 = macosx_deployment_target.replace(".", "_")
+        elif "MACOSX_DEPLOYMENT_TARGET" in os.environ:
```

Comment:
Why are we checking this? It would somehow feel a lot cleaner if tests don't check the environment but always make sure they get the same output?

Reply:
I do not use `"MACOSX_DEPLOYMENT_TARGET` in test. I introduce this if to test not fail if `"MACOSX_DEPLOYMENT_TARGET` is defined in user environment. 
Other approach is to modify environment in test call to be sure  that this environment variable is not defined.

Reply:
Hmmm, myeah, I see. I'm not entirely convinced, but it would be nice to be independent from the environment, no? I guess in that case, we'd add a line like `env.pop('MACOSX_DEPLOYMENT_TARGET', None)` in `run_cibuildwheel`?

Reply:
OK, I've tried this. Let's see what the CIs say :-)

Reply:
In your commit there is no removing `MACOSX_DEPLOYMENT_TARGET` from environment.  

the c++ standards tests are not problematic because `MACOSX_DEPLOYMENT_TARGET` is set inside. Problem is with all other tests.

Reply:
@Czaki, I did, no?

https://github.com/joerick/cibuildwheel/pull/156/commits/fc67e3db947e3ba9186fe39d1010d47300101fbe#diff-8437219d14af4459b32737acc9521403R57

Reply:
My bad. I removed this commit.

Reply:
But tests still fail? :-(

I don't have rights on CircleCI, I'm afraid. :-/ Maybe amend the commit without changes and force push again?

Reply:
It is canceled because I push next commit. 

I restart all tests with force push.

Reply:
Great, let's see :-)

Reply:
Do you agree with these changes, btw? Or do you consider the previous solution to be better?
I mainly just wanted to try this out, and see if it would look better; we can still revert it if you want.

Reply:
I agree. It is better, because it still check version forced by choosing python installer and reduce change of behavior tests between environments.

Reply:
OK, great, then! :-) Then we mainly just have to wait for a new wheel release. Wondering if that's getting close; should be, I hope.

Reply:
I see one problem. If there will be a test that will run cibuildwheel in different way than using function from `utils` then this protection will not be apply.  But maybe it needs only mention in Readme?

Reply:
I wouldn't put it in the README. This is more of a development thing, right?

I wouldn't really know where to mention this, actually. Maybe we should just solve the problem when/if it occurs in some PR?

Reply:
It think about this readme https://github.com/joerick/cibuildwheel/blob/master/test/README.md

other approach is to put something like this: 
```
@pytest.fixture(autouse=True)
def no_macosx_deployment_target(monkeypatch):
    """remove MACOSX_DEPLOYMENT_TARGET from environment"""
    if "MACOSX_DEPLOYMENT_TARGET" in os.environ:
        env = os.environ.copy()
        del env["MACOSX_DEPLOYMENT_TARGET"]
        monkeypatch.setattr(os, "environ", env)
```
in conftest.py 

But maybe keep in memory this edge case is the best solution.

Reply:
Oh, I had no idea that was there!
Well, maybe. But I think normally, to make a new test, you copy the simple test `01_basic` anyway, and start working from there?

The one thing I was unsure about was if someone passes `env`. But then again, worst case, tests break and we'll fix something. It's still much better to have a test fail when there's no error, than to have them succeed when there is an error.

Reply:
Actually, I do like your suggestion/solution with the fixture as well. But it's a bit more hidden away because it's in `conftest.py`

Reply:
Ok. I'm quite new in such projects and not sure how strong protection should be.  I'm ok with all proposition.